### ISSUE TITLE
Class Summary Page feature requests

### ISSUE THREAD

**Issue by MaxPat931** [foundryvtt/dnd5e#2125]
1 Should it validate that the subclasses added to the sheet have a matching Class identifier to the Class?

2 Include Optional advancements in the generated class table, currently they are not shown on the Class table but have a separate section and custom table at the bottom of the generated item details

3 The class table lacks identifying when a subclass feature is granted, perhaps once the subclass items are added it can state "[Subclass Header] Feature" in plain text in the table
![image](https://user-images.githubusercontent.com/86370342/215292899-1a54b457-7783-494c-b1be-4a37c8acf503.png)

4 Special Handling for Item Grant advancements with only Spell Items
Currently if you use the advancement system to grant Prepared spellcasters spells, the table and item details sections balloon quite a lot, perhaps showing an inline link for the advancement item in the class table or just plain text, then in the description section a table to show the links to the spells similar to the optional display (but without the description of the individual spell items)
Current:
![image](https://user-images.githubusercontent.com/86370342/215292421-e097af87-6d67-41b9-a2df-005325c22518.png)
Proposed:
![image](https://user-images.githubusercontent.com/86370342/215292707-85267ecf-8d81-4d10-96de-7cdf28f71201.png)
![image](https://user-images.githubusercontent.com/86370342/215292318-a2c84ee5-110d-4ef9-ac09-2ef01ca38ffa.png)

Thanks

**Comment by arbron** [comment]
> 1 Should it validate that the subclasses added to the sheet have a matching Class identifier to the Class?

I don't think this is too much of an issue, but we can always take another look at that when `SubclassAdvancement` is added.

> 2 Include Optional advancements in the generated class table, currently they are not shown on the Class table but have a separate section and custom table at the bottom of the generated item details

None of the official sources mix required features with optional features in the same table, so I've taken the route that D&D Beyond uses when they show their combined class summary pages.

> 3 The class table lacks identifying when a subclass feature is granted, perhaps once the subclass items are added it can state "[Subclass Header] Feature" in plain text in the table ![image](https://user-images.githubusercontent.com/86370342/215292899-1a54b457-7783-494c-b1be-4a37c8acf503.png)

We don't have this information at the moment, but it was something I was planning when `SubclassAdvancement` is added. Right now subclass handling is more of a stopgap design until that advancement is in place.

> 4 Special Handling for Item Grant advancements with only Spell Items Currently if you use the advancement system to grant Prepared spellcasters spells, the table and item details sections balloon quite a lot, perhaps showing an inline link for the advancement item in the class table or just plain text, then in the description section a table to show the links to the spells similar to the optional display (but without the description of the individual spell items) Current: ![image](https://user-images.githubusercontent.com/86370342/215292421-e097af87-6d67-41b9-a2df-005325c22518.png) Proposed: ![image](https://user-images.githubusercontent.com/86370342/215292707-85267ecf-8d81-4d10-96de-7cdf28f71201.png) ![image](https://user-images.githubusercontent.com/86370342/215292318-a2c84ee5-110d-4ef9-ac09-2ef01ca38ffa.png)

Spells granted are not something normally shown on class summaries. Perhaps `ItemGrantAdvancement`s can be given a toggle as to whether they are shown in the table at all, but any further improvement to this will probably wait until spell lists are properly integrated into the system.

**Comment by JPMeehan** [comment]
Until spell lists are added to the system I would appreciate advancements that only have spell items being suppressed from the class summary - lots of subclass spell lists (Artificer Specializations, Cleric Domains, etc.) can be reasonably modeled as Item Grant advancements with the current Always Prepared model, the only deficiency is on the Class Summary sheet where it's now giving me the full text of _flaming sphere_.

**Comment by flogiston1** [comment]
I would like to add one more feature request: If the tables in class/subclass summary pages could be disabled altogether with some toggle. The tables are great, but there are some cases I'd like the option to hide some of them.

I have formatted subclass items so that they have tables in their description, and now I get subclass tables twice. Sometimes even a domain spells table or such between them. I could remove the tables from the items but I'd prefer to have them there and disable the generated subclass tables in class summaries.

Also some scale values in subclass items make the tables weird. Example of this can be seen in SRD life domain cleric. Some toggle to hide specific things in tables would be nice too.

**Comment by krbz999** [comment]
- It is currently difficult to justify using subclass pages in premium content for subclasses that grant spells.
- In addition, it would be good if the automatically generated subclass table did not show rows that had no changes from the previous level. As is, if a subclass has a scale value, it will show its value at all 20 levels, not just those levels with actual changes.
